packages/ra-ui-materialui/src/input/common/dataProvider.tsx

```diff
@@ -0,0 +1,246 @@
+                },
+                {
+                    name: 'Alexander Pushkin',
+                    role: 'co_writer',
+                },
+            ],
+            author_gender: 'F',
+            year: 1954,
+            time: '2024-02-12T13:45:29.170Z',
+            genre: [7, 8],
+            published: true,
+        },
+        {
+            id: 8,
+            title: {
+                en: "Harry Potter and the Philosopher's Stone",
+                fr: 'Harry Potter et la pierre philosophale',
+            },
+            authors: [
+                {
+                    name: 'J. K. Rowling',
+                    role: 'head_writer',
+                },
+                {
+                    name: 'Alexander Pushkin',
+                    role: 'co_writer',
+                },
+            ],
+            author_gender: 'F',
+            year: 1997,
+            time: '2024-02-12T13:45:29.170Z',
+            genre: [7, 8],
+            published: true,
+        },
+        {
+            id: 9,
+            title: {
+                en: 'The Alchemist',
+                fr: "L'alchimiste",
+            },
+            authors: [
+                {
+                    name: 'Paulo Coelho',
+                    role: 'head_writer',
+                },
+                {
+                    name: 'Alexander Pushkin',
+                    role: 'co_writer',
+                },
+            ],
+            author_gender: 'M',
+            year: 1988,
+            time: '2024-02-12T13:45:29.170Z',
+            genre: [12, 13],
+            published: true,
+        },
+        {
+            id: 10,
+            title: {
+                en: 'A Catcher in the Rye',
+                fr: 'Un receveur dans le seigle',
+            },
+            authors: [
+                {
+                    name: 'J. D. Salinger',
+                    role: 'head_writer',
+                },
+                {
+                    name: 'Alexander Pushkin',
+                    role: 'co_writer',
+                },
+            ],
+            author_gender: 'M',
+            year: 1951,
+            time: '2024-02-12T13:45:29.170Z',
+            genre: [14, 15],
+            published: true,
+        },
+    ],
+    genres: [
+        { id: 1, name: 'Historical Fiction' },
+        { id: 2, name: 'War' },
+        { id: 3, name: 'Classic' },
+        { id: 4, name: 'Romance' },
+        { id: 5, name: 'Gothic' },
+        { id: 6, name: 'Philosophical' },
+        { id: 7, name: 'Fable' },
+        { id: 8, name: 'Fantasy' },
+        { id: 9, name: 'Adventure' },
+        { id: 10, name: 'Literary Realism' },
+        { id: 11, name: 'Tragedy' },
+        { id: 12, name: 'Philosophical' },
+        { id: 13, name: 'Adventure' },
+        { id: 14, name: 'Coming-of-age' },
+        { id: 15, name: 'Literary Fiction' },
+    ],
+    authors: [],
+};
+
+export const dataProvider = fakeRestDataProvider(data, true);
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** To avoid logging during the automated tests

‏
```suggestion
export const dataProvider = fakeRestDataProvider(data, process.env.NODE_ENV !== 'test');
```